lib/restrictor.cu

```diff
@@ -1,145 +1,109 @@
 #include <color_spinor_field.h>
-#include <tune_quda.h>
-#include <launch_kernel.cuh>
-
-#include <jitify_helper.cuh>
+#include <tunable_block_reduction.h>
 #include <kernels/restrictor.cuh>
 
 namespace quda {
 
-  template <typename Float, typename vFloat, int fineSpin, int fineColor, int coarseSpin, int coarseColor,
-            int coarse_colors_per_thread>
-  class RestrictLaunch : public Tunable {
+  struct Aggregates {
+    // List of block sizes we wish to instantiate.  The required block
+    // size is equal to number of fine points per aggregate, rounded
+    // up to a whole power of two.  So for example, 2x2x2x2 and
+    // 3x3x3x1 aggregation would both use the same block size 32
+#ifndef QUDA_FAST_COMPILE_REDUCE
+    static constexpr std::array<unsigned int, 6> block = {32, 64, 128, 256, 512, 1024};
+#else
+    static constexpr std::array<unsigned int, 1> block = {1024};
+#endif
+
+    /**
```

Comment:
just wondering: This is duplicated from block_orthogonalize.cu. Can we avoid that?